src/backups.py

```diff
@@ -1303,7 +1306,18 @@ def start_stop_pgbackrest_service(self) -> bool:
             return False
 
         # Start the service.
-        self.container.restart(self.charm.pgbackrest_server_service)
+        services = self.container.pebble.get_services(names=[self.charm.pgbackrest_server_service])
+        if len(services) == 0:
+            return False
+
+        if services[0].current == ServiceStatus.ACTIVE:
+            logger.debug("Sending SIGHUP to pgBackRest TLS server to reload configuration")
+            self.container.pebble.send_signal(
+                signal.SIGHUP, services=[self.charm.pgbackrest_server_service]
+            )
```

Comment:
Always restarting the service was causing a SIGTERM to be sent to the pgBackRest TLS  server in the middle of the process where the replica gets the latest data from the primary to create a backup. This started to happen because of some more deferred events happening. A SIGHUP is enough to update the TLS server when the certificates change.